src/documents/models.py

```diff
@@ -483,8 +483,14 @@ def from_path(cls, path):
           "<title>.<suffix>"
         """
 
+        filename = os.path.basename(path)
```

Comment:
This is crafty, I like it.  Can you add a comment here about what's going on though?  Just for future nerds wanting to make changes.  I stared at this for about 5min until I figured it out.  In my case I hadn't realised that you'd taken an actual filename (from line 486) and replaced it with a transformed filename (488).  Just some a line or two that explains what's going on would be handy.

Reply:
How does this sound to you as a comment?

> Apply at most one transformation configured in
> FILENAME_PARSE_TRANSFORMS to filename, before it is
> consumed by the parser. This allows to adapt custom
> filename formats to what paperless expects.
> (see [...])